src/pt/flowermanga/src/eu/kanade/tachiyomi/extension/pt/flowermanga/FlowerManga.kt

```diff
@@ -1,22 +1,59 @@
 package eu.kanade.tachiyomi.extension.pt.flowermanga
 
+import android.app.Application
+import android.content.SharedPreferences
+import androidx.preference.ListPreference
+import androidx.preference.PreferenceScreen
 import eu.kanade.tachiyomi.multisrc.madara.Madara
 import eu.kanade.tachiyomi.network.interceptor.rateLimit
+import eu.kanade.tachiyomi.source.ConfigurableSource
 import okhttp3.OkHttpClient
+import uy.kohesive.injekt.Injekt
+import uy.kohesive.injekt.api.get
 import java.text.SimpleDateFormat
 import java.util.Locale
 import java.util.concurrent.TimeUnit
 
-class FlowerManga : Madara(
-    "Flower Manga",
-    "https://flowermanga.net",
-    "pt-BR",
-    SimpleDateFormat("d 'de' MMMMM 'de' yyyy", Locale("pt", "BR")),
-) {
+class FlowerManga :
+    Madara(
+        name = "Flower Manga",
+        lang = "pt-BR",
+        baseUrl = MIRRORS[DEFAULT_MIRROR],
+        dateFormat = SimpleDateFormat("d 'de' MMMMM 'de' yyyy", Locale("pt", "BR")),
+    ),
+    ConfigurableSource {
+
     override val client: OkHttpClient = super.client.newBuilder()
         .rateLimit(1, 2, TimeUnit.SECONDS)
         .build()
 
     override val useLoadMoreRequest = LoadMoreStrategy.Never
+
     override val useNewChapterEndpoint = false
+
+    override val baseUrl: String by lazy {
+        val index = preference.getString(MIRROR_KEY, "$DEFAULT_MIRROR")!!.toInt().coerceAtMost(MIRRORS.size - 1)
```

Comment:
Should add all URLs for CI, see https://github.com/search?q=repo%3Akeiyoushi%2Fextensions-source+%2FjoinToString%5C%28%5C%22%23%2C+%5C%22%5C%29%2F&type=code